subsys/zigbee/cli/zigbee_cli.c

```diff
@@ -0,0 +1,149 @@
+
+
+/**@brief Function for CLI shell initial configuration.
+ *        To be called by SYS_INIT.
+ *
+ * @param[in] unused Unused param.
+ *
+ * @returns By default returns 0.
+ */
+static int zb_cli_init(struct device *unused)
+{
+	ARG_UNUSED(unused);
+
+#ifdef CONFIG_ZIGBEE_CLI_SHELL_PROMPT
+	zb_set_cli_default_shell_prompt(CONFIG_ZIGBEE_CLI_SHELL_PROMPT);
+#endif
+
+	return 0;
+}
+
+/**@brief Function for setting prompt for CLI shell. Omits empty string.
+ *
+ * @param[in] new_prompt  Pointer to new CLI shell prompt.
+ */
+void zb_set_cli_default_shell_prompt(const char *new_prompt)
+{
+	if (strlen(new_prompt) < 1) {
+		return;
+	}
+#ifdef CONFIG_SHELL_BACKEND_SERIAL
+	if (shell_prompt_change(shell_backend_uart_get_ptr(), new_prompt)) {
+		LOG_ERR("Can not change shell prompt");
+	}
+#endif
+#ifdef CONFIG_SHELL_BACKEND_RTT
+	if (shell_prompt_change(shell_backend_rtt_get_ptr(), new_prompt)) {
+		LOG_ERR("Can not change shell prompt");
+	}
+#endif
+}
+
+/**@brief Returns the Endpoint number used by the CLI.
+ */
+zb_uint8_t zb_get_cli_endpoint(void)
+{
+	return cli_ep;
+}
+
+/**@brief Sets the Endpoint number used by the CLI.
+ */
+zb_void_t zb_set_cli_endpoint(zb_uint8_t ep)
+{
+	cli_ep = ep;
+}
+
+/**@brief Configures CLI endpoint by setting number of endpoint to be used
+ * by the CLI and for chosen endpoint registers CLI endpoint handler.
+ */
+zb_void_t zb_cofigure_cli_endpoint(void)
+{
+	zb_zcl_globals_t *zcl_ctx = zb_zcl_get_ctx();
+	zb_af_endpoint_desc_t *cli_ep_desc;
+
+	/* Verify that ZCL device ctx is present at the device as for example,
+	 * network coordinator sample doesn't have one by default.
+	 */
+	if (zcl_ctx->device_ctx == NULL) {
+		LOG_ERR("Register ZCL ctx at Zigbee device to use CLI ZCL commands.");
+		return;
+	}
+
+	/* Set endpoint number to use from KConfig. */
+#ifdef CONFIG_ZIGBEE_CLI_ENDPOINT
+	zb_set_cli_endpoint(CONFIG_ZIGBEE_CLI_ENDPOINT);
+#endif
+	/* Verify that endpoint used by CLI is registered. */
+	cli_ep_desc = zb_af_get_endpoint_desc(zb_get_cli_endpoint());
+	if (!cli_ep_desc) {
+		LOG_ERR("CLI endpoint: %d is not registered.",
+			zb_get_cli_endpoint());
+		return;
+	}
+
+	/* Check that no endpoint device handler is set for CLI endpoint
+	 * and set handler used by CLI.
+	 */
+	if (cli_ep_desc->device_handler) {
+		LOG_ERR("Can not set device handler for CLI endpoint - handler already set.");
+	} else {
+		ZB_AF_SET_ENDPOINT_HANDLER(zb_get_cli_endpoint(),
+					   cli_agent_ep_handler);
+	}
+}
+
+/**@brief Sets the debug mode.
+ */
+zb_void_t zb_cli_debug_set(zb_bool_t debug)
+{
+	debug_mode = debug;
+}
+
+/**@brief Gets the debug mode.
+ */
+zb_bool_t zb_cli_debug_get(void)
+{
+	return debug_mode;
+}
+
+/**@brief Function for getting the state of the Zigbee stack
+ *        processing suspension.
+ */
+zb_bool_t zb_cli_stack_is_suspended(void)
+{
+	k_tid_t zboss_thread = zb_get_zboss_thread_id();
+
+	if (zboss_thread) {
+		if (!(zboss_thread->base.thread_state & _THREAD_SUSPENDED)) {
```

Comment:
Perhaps there is an API to get thread state? Seems like you are accessing some private data here.